src/openforms/submissions/pricing.py

```diff
@@ -32,8 +48,32 @@ def get_submission_price(submission: "Submission") -> Decimal:
     ), "get_submission_price' may only be called for forms that require payment"
 
     form = submission.form
-    data = submission.data
 
+    #
+    # 1. If a variable key is configured, try that.
+    #
+    try:
+        price = _price_from_variable(submission)
+    except InvalidPrice as exc:
```

Comment:
When would this be raised?In `_price_from_variable` I see we return `None`, `Decimal(value)` and otherwise we raise an exception. I am a bit confused here, I thought that this exception would be raised before we reach this point and I do not see a test for this tbh. I guess I m missing something here..

Reply:
E.g. `Decimal("")` results in `decimal.InvalidOperation` in `_price_from_variable`, which is caught and re-raised as `InvalidPrice`

While the UI should only allow selecting variables of numeric type, I don't doubt for a moment that someone will mess with the JSON and end up with prices pointing to a string variable which will be unset/not initialized properly due to broken form logic, and that's why this defensive programming is here - I want to make those error cases explicit as quickly as possible instead of letting the problem bubble through the system.